lib/ansible/utils/collection_loader.py

```diff
@@ -47,6 +49,12 @@ def __init__(self, config=None):
         elif self._n_configured_paths is None:
             self._n_configured_paths = []
 
+        # Append all ``ansible_collections`` dirs from sys.path to the end
```

Comment:
This comment would be a bit more useful if it explained the effect of the change:
```suggestion
        # Append all ``ansible_collections`` dirs from sys.path to the end
        # which enables the collection loader to locate and import Python
        # modules that are distributed as Python distributed packages via
        # PyPI or similar indexes. Specifically, it enables users to distribute
        # Python content as long as dists follow PEP 420 Implicit Namespace
        # Packages conventions.
        # Refs:
        # * https://packaging.python.org/guides/packaging-namespace-packages/
        # * https://packaging.python.org/guides/creating-and-discovering-plugins/#using-namespace-packages
        # * https://setuptools.readthedocs.io/en/latest/setuptools.html#namespace-packages
        # * https://setuptools.readthedocs.io/en/latest/setuptools.html#find-namespace-packages
        #
        # SECURITY CONSIDERATIONS:
        # This provides yet another place for injecting Python code.
        # We should be clear about it when informing users about the places
        # where modules/plugins are loaded from.
        #
        # NOTE: It is possible to publish a "broken" namespace package
        # NOTE: so that when it's installed it breaks user's ability
        # NOTE: to install other Python packages in that namespace.
```

Reply:
I'd take out the REFS.  Security considerations should go somewhere else.  maybe there should be a single place in the documentation which lists all the locations that the collections can be loaded from.  The note is correct but within ansible, we use our special loader which establishes a namespace prior to loading anything else so this should only break with third party code which might try to load the ansible-collections package without establishing the namespace first.  So if we state that here, it should note that it won't break us, only third parties.